src/twisted/conch/test/test_filetransfer.py

```diff
@@ -535,18 +535,23 @@ def test_openDirectoryIterator(self):
 
         d = self.client.openDirectory(b"")
         self._emptyBuffers()
-        openDir = yield d
+        openDir = await d
 
         filenames = set()
         try:
             for f in openDir:
                 self._emptyBuffers()
-                (filename, _, fileattrs) = yield f
+                v = await f if isinstance(f, Awaitable) else f
+                # PEP492 (async/await) implies PEP479 (generator_stop)
+                if v is None and f.result.check(StopIteration):
+                    break
```

Comment:
Would testing for `Deferred` be sufficient here? Testing for `Awaitable` suggests the code is very generic, but `f.result` does not work with other awaitable types.